bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/JSONParser.java

```diff
@@ -107,8 +128,9 @@ private static Object changeForBString(Object jsonObj) {
      * @return JSON structure
      * @throws BallerinaException for any parsing error
      */
-    public static Object parse(Reader reader) throws BallerinaException {
+    public static Object parse(Reader reader, NonStringValueProcessingMode mode) throws BallerinaException {
         StateMachine sm = tlStateMachine.get();
+        sm.mode = mode;
```

Comment:
Still not sure if we can set the mode directly to the state machine even though it is a thread local variable since we can have a scenario with multiple workers on the same thread?

Reply:
Looking at `reset` method I don't believe parser is created to be used by two workers on same thread.
https://github.com/ballerina-platform/ballerina-lang/blob/4babdbae9864c55e33be2e2a41b469175fdd950d/bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/JSONParser.java#L202

Reply:
Yeah, we'll have to verify this and introduce changes if required. 

I guess we can go ahead with Chiran's changes for the time-being if this is an issue in general, but let's verify first.

Reply:
Ballerina schedule will reuse the same thread for multiple workers, but it uses a non-preemptive strategy. That is, we only yield and resume strands at given points. Since there are no resume points inside extern  code, java method should execute from the top each time (ie, atomic). 
Therefor, thread local state should be safe.

Reply:
@chiranSachintha any workaround for adhering to this breaking API change. And please add the API doc description for this new param.

Reply:
> @chiranSachintha any workaround for adhering to this breaking API change. And please add the API doc description for this new param.

Fixed with this PR (https://github.com/ballerina-platform/ballerina-lang/pull/25881)